Failed to support gradle splits mechanism

I'm using the splits mechanism that the Google Android Gradle Plugin is offering to generate a version of my app for each processor architecture.
When I apply the android-unit-test plugin, I get the error below.
Any chances to fix that issue sooner or later?

Here is the stack trace: 

Caused by: java.lang.RuntimeException: More than one output on variant cannot call getProcessManifest() on it. Call it on one of its outputs instead.
    at com.android.build.gradle.internal.api.BaseVariantImpl.getProcessManifest(BaseVariantImpl.java:297)
    at com.android.build.gradle.internal.api.ApkVariantImpl.getProcessManifest(ApkVariantImpl.java:40)
    at com.android.build.gradle.internal.api.ApplicationVariantImpl_Decorated.getProcessManifest(Unknown Source)
    at org.gradle.api.internal.BeanDynamicObject$MetaClassAdapter.getProperty(BeanDynamicObject.java:153)
    at org.gradle.api.internal.BeanDynamicObject.getProperty(BeanDynamicObject.java:107)
    at org.gradle.api.internal.CompositeDynamicObject.getProperty(CompositeDynamicObject.java:78)
    at com.android.build.gradle.internal.api.ApplicationVariantImpl_Decorated.getProperty(Unknown Source)
    at com.jcandksolutions.gradle.androidunittest.VariantWrapper.getMergedManifest(VariantWrapper.groovy:118)
    at com.jcandksolutions.gradle.androidunittest.VariantWrapper.getProperty(VariantWrapper.groovy)
    at com.jcandksolutions.gradle.androidunittest.TaskManager.createTestTask(TaskManager.groovy:83)
    at com.jcandksolutions.gradle.androidunittest.TaskManager$createTestTask.call(Unknown Source)
    at com.jcandksolutions.gradle.androidunittest.MainHandler$_run_closure1.doCall(MainHandler.groovy:50)
    at org.gradle.api.internal.ClosureBackedAction.execute(ClosureBackedAction.java:59)
    at org.gradle.listener.ActionBroadcast.execute(ActionBroadcast.java:39)
    at org.gradle.api.internal.DefaultDomainObjectCollection.doAdd(DefaultDomainObjectCollection.java:164)
    at org.gradle.api.internal.DefaultDomainObjectCollection.add(DefaultDomainObjectCollection.java:159)
    at com.android.build.gradle.AppExtension.addVariant(AppExtension.groovy:48)
    at com.android.build.gradle.internal.VariantManager.createVariantApiObjects(VariantManager.java:609)
    at com.android.build.gradle.internal.VariantManager.createApiObjects(VariantManager.java:576)
    at com.android.build.gradle.internal.VariantManager.createAndroidTasks(VariantManager.java:222)
    at com.android.build.gradle.internal.VariantManager$createAndroidTasks$0.call(Unknown Source)
    at com.android.build.gradle.BasePlugin.createAndroidTasks(BasePlugin.groovy:470)
    at com.android.build.gradle.BasePlugin$_createTasks_closure9.doCall(BasePlugin.groovy:407)
    at org.gradle.listener.ClosureBackedMethodInvocationDispatch.dispatch(ClosureBackedMethodInvocationDispatch.java:40)
    at org.gradle.listener.ClosureBackedMethodInvocationDispatch.dispatch(ClosureBackedMethodInvocationDispatch.java:25)
    at org.gradle.listener.BroadcastDispatch.dispatch(BroadcastDispatch.java:83)
    at org.gradle.listener.BroadcastDispatch.dispatch(BroadcastDispatch.java:31)
    at org.gradle.messaging.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
    at com.sun.proxy.$Proxy13.afterEvaluate(Unknown Source)
    at org.gradle.configuration.project.LifecycleProjectEvaluator.notifyAfterEvaluate(LifecycleProjectEvaluator.java:67)
    ... 30 more

--- a/.travis.yml
+++ b/.travis.yml
@@ -4,7 +4,7 @@ android:
   components:
     - platform-tools
     - tools
-    - build-tools-21.1.0
+    - build-tools-21.1.1
     - android-21
     - extra-android-m2repository
     - extra-google-m2repository

--- a/example/exampleApp/build.gradle
+++ b/example/exampleApp/build.gradle
@@ -5,7 +5,7 @@ buildscript {
       mavenLocal()
     }
 
-    classpath 'com.android.tools.build:gradle:0.14.0'
+    classpath 'com.android.tools.build:gradle:0.14.4'
     classpath 'com.github.jcandksolutions.gradle:android-unit-test:2.0.2-SNAPSHOT'
   }
 }
@@ -14,11 +14,17 @@ apply plugin: 'com.android.application'
 
 android {
   compileSdkVersion 21
-  buildToolsVersion '21.1.0'
+  buildToolsVersion '21.1.1'
 
   buildTypes {
     debug {
       applicationIdSuffix ".debug"
+    }
+  }
+
+  splits {
+    abi {
+      enable true
     }
   }
 

--- a/example/exampleApp/exampleApp.iml
+++ b/example/exampleApp/exampleApp.iml
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<module external.linked.project.id=":exampleApp" external.linked.project.path="$MODULE_DIR$" external.root.project.path="$MODULE_DIR$/.." external.system.id="GRADLE" external.system.module.group="example" external.system.module.version="unspecified" type="JAVA_MODULE" version="4">
+<module external.linked.project.id=":exampleApp" external.system.module.group="example" external.system.module.version="unspecified" type="JAVA_MODULE" version="4">
   <component name="FacetManager">
     <facet type="android-gradle" name="Android-Gradle">
       <configuration>
@@ -106,7 +106,7 @@
       <excludeFolder url="file://$MODULE_DIR$/build/test-resources" />
       <excludeFolder url="file://$MODULE_DIR$/build/test-results" />
     </content>
-    <orderEntry type="jdk" jdkName="Android API 20 Platform" jdkType="Android SDK" />
+    <orderEntry type="jdk" jdkName="Android API 21 Platform" jdkType="Android SDK" />
     <orderEntry type="sourceFolder" forTests="false" />
     <orderEntry type="module" module-name="exampleLibrary" exported="" />
   </component>

--- a/example/exampleLibrary/build.gradle
+++ b/example/exampleLibrary/build.gradle
@@ -5,7 +5,7 @@ buildscript {
       mavenLocal()
     }
 
-    classpath 'com.android.tools.build:gradle:0.14.0'
+    classpath 'com.android.tools.build:gradle:0.14.4'
     classpath 'com.github.jcandksolutions.gradle:android-unit-test:2.0.2-SNAPSHOT'
   }
 }
@@ -14,7 +14,7 @@ apply plugin: 'com.android.library'
 
 android {
   compileSdkVersion 21
-  buildToolsVersion '21.1.0'
+  buildToolsVersion '21.1.1'
 
   // THIS IS VERY IMPORTANT! THIS IS THE PACKAGE UNDER
   // WHICH THE R CLASS IS GENERATED. IT SHOULD BE THE

--- a/example/exampleLibrary/exampleLibrary.iml
+++ b/example/exampleLibrary/exampleLibrary.iml
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<module external.linked.project.id=":exampleLibrary" external.linked.project.path="$MODULE_DIR$" external.root.project.path="$MODULE_DIR$/.." external.system.id="GRADLE" external.system.module.group="example" external.system.module.version="unspecified" type="JAVA_MODULE" version="4">
+<module external.linked.project.id=":exampleLibrary" external.system.module.group="example" external.system.module.version="unspecified" type="JAVA_MODULE" version="4">
   <component name="FacetManager">
     <facet type="android-gradle" name="Android-Gradle">
       <configuration>
@@ -66,7 +66,7 @@
       <excludeFolder url="file://$MODULE_DIR$/build/test-results" />
       <excludeFolder url="file://$MODULE_DIR$/build/tmp" />
     </content>
-    <orderEntry type="jdk" jdkName="Android API 20 Platform" jdkType="Android SDK" />
+    <orderEntry type="jdk" jdkName="Android API 21 Platform" jdkType="Android SDK" />
     <orderEntry type="sourceFolder" forTests="false" />
   </component>
 </module>

--- a/src/main/groovy/com/jcandksolutions/gradle/androidunittest/VariantWrapper.groovy
+++ b/src/main/groovy/com/jcandksolutions/gradle/androidunittest/VariantWrapper.groovy
@@ -115,7 +115,7 @@ public abstract class VariantWrapper {
    */
   public File getMergedManifest() {
     if (mMergedManifest == null) {
-      mMergedManifest = mVariant.processManifest.manifestOutputFile
+      mMergedManifest = mVariant.outputs.first().processManifest.manifestOutputFile
     }
     return mMergedManifest
   }

--- a/src/test/groovy/com/jcandksolutions/gradle/androidunittest/VariantWrapperTest.groovy
+++ b/src/test/groovy/com/jcandksolutions/gradle/androidunittest/VariantWrapperTest.groovy
@@ -2,6 +2,7 @@ package com.jcandksolutions.gradle.androidunittest
 
 import com.android.build.gradle.api.ApplicationVariant
 import com.android.build.gradle.api.BaseVariant
+import com.android.build.gradle.api.BaseVariantOutput
 import com.android.build.gradle.tasks.ManifestProcessorTask
 import com.android.build.gradle.tasks.MergeAssets
 import com.android.builder.core.DefaultBuildType
@@ -70,6 +71,8 @@ public class VariantWrapperTest {
     mSourceSet = mock(SourceSet.class)
     mVariant = mock(ApplicationVariant.class)
     mMergedManifest = mock(File.class)
+    List<BaseVariantOutput> outputs = new ArrayList<>()
+    BaseVariantOutput output = mock(BaseVariantOutput.class)
     ManifestProcessorTask manTask = mock(ManifestProcessorTask.class)
     DefaultBuildType buildType = mock(DefaultBuildType.class)
     mResources = mock(SourceDirectorySet.class)
@@ -107,7 +110,9 @@ public class VariantWrapperTest {
     when(free.name).thenReturn("free")
     when(paid.name).thenReturn("paid")
     when(mVariant.productFlavors).thenReturn(productFlavors)
-    when(mVariant.processManifest).thenReturn(manTask)
+    when(mVariant.outputs).thenReturn(outputs)
+    outputs.add(output)
+    when(output.processManifest).thenReturn(manTask)
     when(mVariant.buildType).thenReturn(buildType)
     when(mVariant.javaCompile).thenReturn(androidJavaCompileTask)
     when(mVariant.dirName).thenReturn("variantDirName")

